agent/src/com/cloud/agent/Agent.java

```diff
@@ -697,16 +697,34 @@ private Answer setupDirectDownloadCertificate(SetupDirectDownloadCertificate cmd
 
         final String keyStoreFile = agentFile.getParent() + "/" + KeyStoreUtils.KS_FILENAME;
 
-        String cerFile = agentFile.getParent() + "/" + certificateName + ".cer";
-        Script.runSimpleBashScript(String.format("echo '%s' > %s", certificate, cerFile));
+        String cerFile = agentFile.getParent() + "/" + "CSCERTIFICATE-" + certificateName;
 
+        s_logger.debug("Creating temporary certificate file into: " + cerFile);
+        int result = Script.runSimpleBashScriptForExitValue(String.format("echo '%s' > %s", certificate, cerFile));
+        if (result != 0) {
+            return new Answer(cmd, false, "Could not create the certificate file on path: " + cerFile);
+        }
+
+        s_logger.debug("Retrieving keystore password");
         String privatePasswordFormat = "sed -n '/keystore.passphrase/p' '%s' 2>/dev/null  | sed 's/keystore.passphrase=//g' 2>/dev/null";
         String privatePasswordCmd = String.format(privatePasswordFormat, agentFile.getAbsolutePath());
         String privatePassword = Script.runSimpleBashScript(privatePasswordCmd);
+        if (org.apache.commons.lang.StringUtils.isBlank(privatePassword)) {
```

Comment:
we can't use an unprotected keystore on a private system?

Reply:
Actually the check is because we're using the existing cloud.jks keystore and it is password protected